optuna/integration/lightgbm.py

```diff
@@ -16,22 +21,37 @@
     # API from optuna integration.
     from optuna.integration import lightgbm_tuner as tuner
 
-    # Workaround for mypy.
+    # To pass mypy in tests/integration_tests/lightgbm_tuner_tests/test_optimize.py
     from lightgbm import Dataset  # NOQA
     from optuna.integration.lightgbm_tuner import LightGBMTuner  # NOQA
 
-    _names_from_tuners = ['train', 'LGBMModel', 'LGBMClassifier', 'LGBMRegressor']
+    # Attach lightgbm API.
+    _names_from_tuners = ['LGBMModel', 'LGBMClassifier', 'LGBMRegressor']
 
     # API from lightgbm.
     for api_name in lgb.__dict__['__all__']:
         if api_name in _names_from_tuners:
             continue
         setattr(sys.modules[__name__], api_name, lgb.__dict__[api_name])
 
+    # API from lightgbm_tuner.
     for api_name in _names_from_tuners:
         setattr(sys.modules[__name__], api_name, tuner.__dict__[api_name])
-else:
-    LightGBMTuner = object  # type: ignore
+
+
+@experimental("0.18.0")
+def train(*args, **kwargs):
```

Comment:
I'm concerning about a part of `LightGBMTuner` logic (i.e., `train`) is located in the same file of `LightGBMPruningCallback`. This is not a strong opinion, but it may be beneficial if we keep the `train` in `optuna/integration/lightgbm_tuner` in terms of maintenance cost.

For instance, I think we can apply `try-except` to the import lines in `optuna/integration/lightgbm_tuner/__init__.py` as follows:
```diff
diff --git a/optuna/integration/lightgbm.py b/optuna/integration/lightgbm.py
index 9f6dec89..466bd32d 100644
--- a/optuna/integration/lightgbm.py
+++ b/optuna/integration/lightgbm.py
@@ -31,7 +31,8 @@ if _available:
     for api_name in _names_from_tuners:
         setattr(sys.modules[__name__], api_name, tuner.__dict__[api_name])
 else:
-    LightGBMTuner = object  # type: ignore
+    from optuna.integration import lightgbm_tuner as tuner
+    setattr(sys.modules[__name__], 'train', tuner.__dict__['train'])
 
 
 class LightGBMPruningCallback(object):
diff --git a/optuna/integration/lightgbm_tuner/__init__.py b/optuna/integration/lightgbm_tuner/__init__.py
index e81c5c44..94860142 100644
--- a/optuna/integration/lightgbm_tuner/__init__.py
+++ b/optuna/integration/lightgbm_tuner/__init__.py
@@ -1,6 +1,12 @@
 from optuna._experimental import experimental
-from optuna.integration.lightgbm_tuner.sklearn import LGBMClassifier, LGBMModel, LGBMRegressor  # NOQA
-from optuna.integration.lightgbm_tuner.optimize import LightGBMTuner
+try:
+    from optuna.integration.lightgbm_tuner.sklearn import LGBMClassifier, LGBMModel, LGBMRegressor  # NOQA
+    from optuna.integration.lightgbm_tuner.optimize import LightGBMTuner
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # LightGBMTuner is disabled because LightGBM is not available.
+    _available = False
```
(I have not fully checked the code yet, so it will need additional code to make the implementation solid.)

What do you think of this approach?